lib/client.js

```diff
@@ -86,17 +86,69 @@ async function get(config) {
     );
   }
 
-  let jwks;
-  if (config.clientAssertionSigningKey) {
-    const jwk = JWK.asKey(config.clientAssertionSigningKey).toJWK(true);
-    jwks = { keys: [jwk] };
+  const configTokenAuthMethod = config.clientAuthMethod;
+  const issuerTokenAuthMethods = Array.isArray(
+    issuer.token_endpoint_auth_methods_supported
+  )
+    ? issuer.token_endpoint_auth_methods_supported
+    : [];
+  if (!issuerTokenAuthMethods.includes(configTokenAuthMethod)) {
+    debug(
+      'Client authn method %o is not supported by the issuer. ' +
+        'Supported authn methods are %o.',
+      configTokenAuthMethod,
+      issuerTokenAuthMethods
+    );
+  }
+
+  if (config.clientAuthMethod == 'private_key_jwt') {
+    const configTokenAuthAlg = config.clientAssertionConfig.signingKey;
```

Comment:
This would be `signingAlg` - but I would rather avoid requiring the user to specify this if we can